src/node_zlib.cc

```diff
@@ -236,6 +239,40 @@ class ZCtx : public AsyncWrap {
         ctx->err_ = deflate(&ctx->strm_, ctx->flush_);
         break;
       case UNZIP:
+        next_expected_header_byte = ctx->strm_.next_in;
+
+        switch (ctx->gzip_id_bytes_read_) {
```

Comment:
Quick question: would it be necessary to check that `ctx->strm_.avail_in > 0` before trying to read the first byte?